### PR TITLE
Fix/14525: SRS - Some text elements have no font resize behaviour

### PR DESCRIPTION
**Pull request by flxschmidt**
## Description
Fix some issues corresponding **Dynamic Font** behaviour, and different font style attributes within a text string.

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-14525

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/SRSConsent/SRSConsentViewModel.swift [hunk 1]
```diff
@@ -147,4 +160,17 @@ class SRSConsentViewModel {
 	}
 	
 	private var subscriptions = Set<AnyCancellable>()
+	
+	private func acknoledgementBulletPoint(for text: String) -> DynamicCell {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/SRSConsent/SRSConsentViewModel.swift]:
```suggestion
	private func acknowledgementBulletPoint(for text: String) -> DynamicCell {
```

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/SRSConsent/SRSConsentViewModel.swift [hunk 2]
```diff
@@ -70,24 +70,19 @@ class SRSConsentViewModel {
...
-				.space(height: 8.0),
-				.bulletPoint(text: AppStrings.SRSConsentScreen.acknowledgement2, alignment: .legal),
-				.space(height: 8.0),
-				.bulletPoint(text: AppStrings.SRSConsentScreen.acknowledgement3, alignment: .legal),
-				.space(height: 8.0),
-				.bulletPoint(text: AppStrings.SRSConsentScreen.acknowledgement4, alignment: .legal),
-				.space(height: 8.0),
-				.bulletPoint(text: AppStrings.SRSConsentScreen.acknowledgement5, alignment: .legal),
-				.space(height: 8.0),
+				acknoledgementBulletPoint(for: AppStrings.SRSConsentScreen.acknowledgement1),
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/SRSConsent/SRSConsentViewModel.swift]:
```suggestion
				acknowledgementBulletPoint(for: AppStrings.SRSConsentScreen.acknowledgement1),
```

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/SRSConsent/SRSConsentViewModel.swift [hunk 3]
```diff
@@ -113,24 +108,42 @@ class SRSConsentViewModel {
...
+	private var legalExtendedDataDonationBulletPoints: [NSAttributedString] {
 		var points = [NSAttributedString]()
-
-		// highlighted texts
-		let attributes: [NSAttributedString.Key: Any] = [
-			.font: UIFont.preferredFont(forTextStyle: .headline)
+		
+		// For normal texts
+		let fontRegularAttributes: [NSAttributedString.Key: Any] = [
+			.font: UIFont.enaFont(for: .body, weight: .regular)
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/SRSConsent/SRSConsentViewModel.swift]: Please don't explicitly set the weight to regular here and below and let it take from fontWeight based on the type, so if we change the fontWeight in future for type body it should work everywhere. 

`
static func enaFont(for style: ENAFont, weight: UIFont.Weight? = nil, italic: Bool = false) -> UIFont 
{
		return UIFont
			.preferredFont(forTextStyle: style.textStyle)
			.scaledFont(size: style.fontSize, weight: weight ?? style.fontWeight, italic: italic)
}
`

**Review by puneetmahali1 — APPROVED** [review]
LGTM!

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/SRSConsent/SRSConsentViewModel.swift (hunk 2 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/SRSConsent/SRSConsentViewModel.swift]: Eagle Eye 👍

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/SRSConsent/SRSConsentViewModel.swift (hunk 3 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/SRSConsent/SRSConsentViewModel.swift]: Good point, is now removed.

**Review by naveeddotio — APPROVED** [review]

**Pull request merged**